code/game/objects/items/weapons/shards.dm

```diff
@@ -87,6 +87,7 @@
 				H.Weaken(3)
 				if(affecting.take_damage(5, 0))
 					H.UpdateDamageIcon()
+					affecting.embed(src)
```

Comment:
I put this in at the last minute to address a edge case, if you are throw. Into space with a piece of glass, you will be ejected at the same speed.

This means that each turf you cross triggers the stepping on glass.